### PR TITLE
Include party allocation related events in integrity check

### PR DESCRIPTION
**Pull request by dasormeter**
Include checks for party related events in IntegrityCheck

### CODE REVIEW

File: ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala [hunk 1]
```diff
@@ -118,6 +118,10 @@ object IntegrityCheck extends App {
         assert(updates.head.isInstanceOf[Update.ConfigurationChanged])
       case Proto.DamlLogEntry.PayloadCase.CONFIGURATION_REJECTION_ENTRY =>
         assert(updates.head.isInstanceOf[Update.ConfigurationChangeRejected])
+      case Proto.DamlLogEntry.PayloadCase.PARTY_ALLOCATION_ENTRY =>
+        assert(updates.head.isInstanceOf[Update.PartyAddedToParticipant])
+      case Proto.DamlLogEntry.PayloadCase.PARTY_ALLOCATION_REJECTION_ENTRY =>
+        assert(updates.head.isInstanceOf[Update.PartyAllocationRejected])
       case _ =>
```
Comment (SamirTalwar) [inline ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala]: Perhaps we could name the ignored cases explicitly? That'd stop people from missing it next time.

Why don't we handle `PACKAGE_UPLOAD_REJECTION_ENTRY`, anyway?

Comment (dasormeter) [reply ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala]: `PACKAGE_UPLOAD_REJECTION_ENTRY` we will handle shortly with @gerolf-da's new and improved take on https://github.com/digital-asset/daml/pull/3391 but atm on master those log entries do not fire Update events so there is nothing yet to assert on

Comment (SamirTalwar) [reply ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala]: Got it, makes sense. It'd still be nice to explicitly state that in the code. The catch-all `case _` seems quite error-prone.

Comment (SamirTalwar) [reply ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala]:
```suggestion
      case Proto.DamlLogEntry.PayloadCase.PACKAGE_UPLOAD_REJECTION_ENTRY =>
        // not handled yet
        ()
      case Proto.DamlLogEntry.PayloadCase.PAYLOAD_NOT_SET =>
```

**Comment by gerolf-da** [comment]
Thanks, @dasormeter!

**Pull request merged**